ui/src/views/WorkerManager/WMEditWorkerPool/index.jsx

```diff
@@ -0,0 +1,224 @@
+import React, { Component } from 'react';
+import { bool } from 'prop-types';
+import TextField from '@material-ui/core/TextField';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import ListSubheader from '@material-ui/core/ListSubheader';
+import MenuItem from '@material-ui/core/MenuItem';
+import CodeEditor from '@mozilla-frontend-infra/components/CodeEditor';
+import ListItem from '@material-ui/core/ListItem';
+import Switch from '@material-ui/core/Switch';
+import { withStyles } from '@material-ui/core';
+import ContentSaveIcon from 'mdi-react/ContentSaveIcon';
+import List from '../../Documentation/components/List';
+import isWorkerTypeNameValid from '../../../utils/isWorkerTypeNameValid';
+import Button from '../../../components/Button';
+import Dashboard from '../../../components/Dashboard';
+
+const gcpConfig = {
+  minCapacity: 0,
+  maxCapacity: 0,
+  capacityPerInstance: 1,
+  machineType: 'n1-highcpu-8',
+  regions: ['us-west2'],
+  userData: {},
+  scheduling: {},
+  networkInterfaces: [{}],
+  disks: [{}],
+};
+const providers = {
+  GCP: 'google',
+};
+
+@hot(module)
+@withStyles(theme => ({
+  successIcon: {
+    ...theme.mixins.successIcon,
+  },
+  createIconSpan: {
+    ...theme.mixins.fab,
+    ...theme.mixins.actionButton,
+  },
+  dropdown: {
+    minWidth: 200,
+  },
+}))
+export default class WMEditWorkerPool extends Component {
+  static defaultProps = {
+    isNewWorkerPool: true,
+  };
+
+  static propTypes = {
+    isNewWorkerPool: bool,
+  };
+
+  state = {
+    workerPool: {
+      name: '',
+      description: '',
+      owner: '',
+      wantsEmail: false,
+      providerType: providers.GCP,
+      providerId: '',
+      config: gcpConfig,
+    },
+    invalidProviderConfig: false,
+  };
+
+  handleInputChange = ({ target: { name, value } }) => {
+    this.setState({ workerPool: { ...this.state.workerPool, [name]: value } });
+  };
+
+  handleSwitchChange = event => {
+    const {
+      target: { value },
+    } = event;
+
+    this.setState({
+      workerPool: {
+        ...this.state.workerPool,
+        [value]: !this.state.workerPool[value],
+      },
+    });
+  };
+
+  handleEditorChange = value => {
+    const { workerPool } = this.state;
+
+    try {
+      workerPool.config = JSON.parse(value);
+
+      this.setState({
+        workerPool,
+        invalidProviderConfig: false,
+      });
+    } catch (err) {
+      workerPool.config = value;
+
+      this.setState({
+        workerPool,
+        invalidProviderConfig: true,
+      });
+    }
+  };
+
+  handleCreateWorkerPool = () => {};
+
+  render() {
+    const { isNewWorkerPool, classes } = this.props;
+    const { workerPool, invalidProviderConfig } = this.state;
+
+    return (
+      <Dashboard
+        title={isNewWorkerPool ? 'Create Worker Pool' : 'Edit Worker Pool'}>
+        <List>
+          <ListItem>
+            <TextField
+              label="Name"
+              name="name"
```

Comment:
"Worker Pool ID"

Reply:
I understood you from the first time, no need to repeat it

Reply:
Ah, sorry, I didn't remember if I'd mentioned it and didn't see a comment at this line.